examples/gallery/plot/points-categorical.py

```diff
@@ -0,0 +1,59 @@
+"""
+Color points by categories
+---------------------------
+The :meth:`pygmt.Figure.plot` method can be used to plot symbols which are
+color-coded by categories.
+"""
+
+import numpy as np
+import pandas as pd
+import pygmt
+
+# Load sample penguins data and convert 'species' column to categorical dtype
+df = pd.read_csv("https://github.com/mwaskom/seaborn-data/raw/master/penguins.csv")
+df["species"] = df.species.astype(dtype="category")
+
+# Use pygmt.info to get region bounds (xmin, xmax, ymin, ymax)
+# The below example will return a numpy array like [2.  4.4 4.3 7.9]
+region = pygmt.info(
+    table=df[["bill_length_mm", "bill_depth_mm"]],  # x and y columns
+    per_column=True,  # report output as a numpy array
+    spacing="3/2",  # rounds x and y intervals by 3 and 2 respectively
+)
+
+# Make our 2D categorical scatter plot, coloring each of the 3 species differently
+fig = pygmt.Figure()
+
+# Generate basemap of 10cm x 10cm size
+fig.basemap(
+    region=region,
+    projection="X10c/10c",
+    frame=['xafg+l"Bill length in mm"', 'yafg+l"Bill depth in mm"', "WSen"],
```

Comment:
Maybe add a title to the plot?

```suggestion
    frame=['xafg+l"Bill length in mm"', 'yafg+l"Bill depth in mm"', 'WSen+t"Penguin size at Palmer Station"'],
```

Reply:
Added.